apps/watchdog/src/display_info.rs

```diff
@@ -0,0 +1,387 @@
+    free_disk_space: u64,
+    tezedge: DiskData,
+    ocaml: DiskData,
+}
+
+impl DiskSpaceData {
+    pub fn to_megabytes(&self) -> Self {
+        Self::new(
+            self.total_disk_space / 1024 / 1024,
+            self.free_disk_space / 1024 / 1024,
+            self.tezedge.clone(),
+            self.ocaml.clone(),
+        )
+    }
+
+    pub fn new(
+        total_disk_space: u64,
+        free_disk_space: u64,
+        tezedge: DiskData,
+        ocaml: DiskData,
+    ) -> Self {
+        Self {
+            total_disk_space,
+            free_disk_space,
+            tezedge,
+            ocaml,
+        }
+    }
+}
+
+impl fmt::Display for DiskSpaceData {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(
+            f,
+            "\nFile system: \n\t Total space: {} MB\n\t Free space: {} MB\n\n{}{}",
+            self.total_disk_space, self.free_disk_space, self.tezedge, self.ocaml,
+        )
+    }
+}
+
+#[derive(Serialize, Debug)]
+pub struct NodeInfo {
+    level: u64,
+    block_hash: String,
+    timestamp: String,
+}
+
+impl NodeInfo {
+    pub fn new(level: u64, block_hash: String, timestamp: String) -> Self {
+        Self {
+            level,
+            block_hash,
+            timestamp,
+        }
+    }
+}
+
+impl fmt::Display for NodeInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(
+            f,
+            "\n\tLevel: {}\n\tBlock hash: {}\n\tTimestamp: {}",
+            self.level, self.block_hash, self.timestamp,
+        )
+    }
+}
+
+#[derive(Serialize)]
+pub struct HeadData {
+    ocaml: NodeInfo,
+    tezedge: NodeInfo,
+}
+
+impl HeadData {
+    pub fn new(ocaml: NodeInfo, tezedge: NodeInfo) -> Self {
+        Self { ocaml, tezedge }
+    }
+}
+
+impl fmt::Display for HeadData {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(
+            f,
+            "\nTezedge node: {}\nOcaml node: {}",
+            self.tezedge, self.ocaml,
+        )
+    }
+}
+
+#[derive(Serialize)]
+pub struct TezedgeSpecificMemoryData {
+    light_node: ProcessMemoryStats,
+    protocol_runners: ProcessMemoryStats,
+}
+
+impl fmt::Display for TezedgeSpecificMemoryData {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut total = ProcessMemoryStats::default();
+        total.merge(self.light_node.clone());
+        total.merge(self.protocol_runners.clone());
+        writeln!(
+            f,
+            "\n\tLight-node: {}\n\tProtocol runners: {}\n\tTotal: {}",
+            self.light_node.to_megabytes(),
+            self.protocol_runners.to_megabytes(),
+            total.to_megabytes()
+        )
+    }
+}
+
+impl TezedgeSpecificMemoryData {
+    pub fn new(light_node: ProcessMemoryStats, protocol_runners: ProcessMemoryStats) -> Self {
+        Self {
+            light_node,
+            protocol_runners,
+        }
+    }
+}
+
+#[derive(Serialize)]
+pub struct MemoryData {
```

Comment:
@adonagy 
I think this could be simplified/unified, ocaml and tezedge looks the same:
```
main_process: ProcessMemoryStats,
sub_processes: Vec<ProcessMemoryStats>
```

less splitting things for ocaml vs tezedge is better